protocol/app/vote_extensions/oracle_client.go

```diff
@@ -0,0 +1,76 @@
+package vote_extensions
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	oracleservicetypes "github.com/skip-mev/slinky/service/servers/oracle/types"
+	"google.golang.org/grpc"
+)
+
+// OracleClient is a wrapper around the default Slinky OracleClient interface. This object is responsible for requesting
+// prices from the x/prices module (originally sent via the sidecar to the price-feed service), and
+// injecting those prices into the vote-extension.
+type OracleClient struct {
+	PricesKeeper PricesKeeper
+}
+
+// NewOracleClient returns a new OracleClient object.
+func NewOracleClient(pricesKeeper PricesKeeper) *OracleClient {
+	return &OracleClient{
+		PricesKeeper: pricesKeeper,
+	}
+}
+
+// Start starts the OracleClient.
+func (o *OracleClient) Start(_ context.Context) error {
+	return nil
+}
+
+// Stop stops the OracleClient.
+func (o *OracleClient) Stop() error {
+	return nil
+}
```

Comment:
shouldn't this contain more logic?

Reply:
We moved the logic for actually querying the sidecar out of the VE handlers and into the daemon. The interface still works as the place in the VE handlers where we get the prices from--just the prices are coming from the keeper here instead of the sidecar.